django/contrib/admin/options.py

```diff
@@ -1379,8 +1381,14 @@ def response_add(self, request, obj, post_url_continue=None):
             self.message_user(request, format_html(msg, **msg_dict), messages.SUCCESS)
             if post_url_continue is None:
                 post_url_continue = obj_url
+            query_string = urlparse(request.build_absolute_uri()).query
+            preserved_qsl = parse_qsl(query_string.replace(preserved_filters, ""))
```

Comment:
Can we move this to a separate hook? Should we also remove `_popup` and `_to_field`?